_posts/2023-09-05-Python-Packaging-as-Publishing.md

```diff
@@ -0,0 +1,300 @@
+
+<!-- excerpt end -->
+
+__TLDR (too long, didn't read);__
+
+Use Pythonic packaging tools and techniques to help avoid [code decay](https://en.wikipedia.org/wiki/Software_rot) and unwanted [code smells](https://en.wikipedia.org/wiki/Code_smell) and increase your development velocity. Increase understanding with unsurprising directory structures like those exhibited in [`pypa/sampleproject`](https://github.com/pypa/sampleproject) or [`scientific-python/cookie`](https://github.com/scientific-python/cookie). Enhance trust by being authentic on source control systems like GitHub ([by customizing your profile](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/about-your-profile)), staying up to date with the [latest supported versions of Python](https://devguide.python.org/versions/), and using security linting tools like [`PyCQA/bandit`](https://github.com/PyCQA/bandit) through [visible + automated GitHub Actions ✅ checks](https://cu-dbmi.github.io/set-website/2023/03/15/Automate-Software-Workflows-with-Github-Actions.html). Connect your projects to others using [`CITATION.cff` files](https://citation-file-format.github.io/), [`CONTRIBUTING.md` files](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors), and using environment + packaging tools like [`poetry`](https://python-poetry.org/docs/) to help others reproduce the same results from your code.
+
+## Why practice packaging?
+
+{% include figure.html image="images/text-vs-book.png" caption="Image comparing how a page with some text and a book look and feel different."  %}
+
+The practice of Python packaging efforts is similar to that of publishing a book. Consider how a bag of text is different from a book. How and _why_ are these things different?
+
+- A book has commonly understood sequencing of content (i.e. copyright page, then title page, then body content pages...).
+- A book often cites references and acknowledges other work explicitly.
+- A book undergoes a manufacturing process which allows the text to be received in many places the same way.
+
+{% include figure.html image="images/python-packaging-to-audience.png" caption="Image showing code undergoing packaging to achieve understanding, trust, and connection for an audience."  %}
+
+These can be thought of metaphors when it comes to packaging in Python. Just like books, software can sometimes have a [code smells](https://en.wikipedia.org/wiki/Code_smell) we want to avoid. Following good packaging practices helps to avoid unwanted code smells while increasing development velocity, maintainability of software through understandability, trustworthiness of the content, and connection to other projects.
```

Comment:
Do books have smells? Maybe elaborate on this analogy, possibly with a sentence description of a code smell?

Reply:
Thanks great point, they do! I'll add in a change to this effect.